src/runtime/EncodedValue.h

```diff
@@ -269,14 +272,14 @@ class EncodedValue {
         const uint8_t tag = *(reinterpret_cast<size_t*>(ptr) + 1);
         if (LIKELY(!(tag & POINTER_VALUE_NOT_OBJECT_TAG_IN_DATA))) {
             return Value(reinterpret_cast<Object*>(ptr));
-        } else if (UNLIKELY(tag == POINTER_VALUE_DOUBLE_TAG_IN_DATA)) {
-            return Value(reinterpret_cast<DoubleInEncodedValue*>(ptr)->value());
+        } else if (UNLIKELY(tag == POINTER_VALUE_NUMBER_TAG_IN_DATA)) {
+            return reinterpret_cast<NumberInEncodedValue*>(ptr)->value();
         } else {
             return Value(reinterpret_cast<PointerValue*>(ptr), Value::FromNonObjectPointer);
         }
 #else
-        if (UNLIKELY(readPointerIsDoubleEncodedValue(ptr))) {
-            return Value(reinterpret_cast<DoubleInEncodedValue*>(ptr)->value());
+        if (UNLIKELY(readPointerIsNumberEncodedValue(ptr))) {
+            return Value(reinterpret_cast<NumberInEncodedValue*>(ptr)->value());
```

Comment:
How about directly returning `NumberInEncodedValue::value()` hear instead of wrapping it with `Value`?

Reply:
Good catch!
I will fix it

Reply:
I've updated my commit